dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ExclusionFileWriter.java

```diff
@@ -0,0 +1,172 @@
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Multimap;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLEventFactory;
+import javax.xml.stream.XMLEventWriter;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.Attribute;
+import javax.xml.stream.events.StartElement;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+
+/** Writer for Linkage Checker exclusion files. */
+class ExclusionFileWriter {
+
+  static final XMLEventFactory eventFactory = XMLEventFactory.newInstance();
+
+  private static final QName LINKAGE_CHECKER_FILTER_TAG = QName.valueOf("LinkageCheckerFilter");
+  private static final QName CLASS_TAG = QName.valueOf("Class");
+  private static final QName LINKAGE_ERROR_TAG = QName.valueOf("LinkageError");
+  private static final QName TARGET_TAG = QName.valueOf("Target");
+  private static final QName SOURCE_TAG = QName.valueOf("Source");
+  private static final QName METHOD_TAG = QName.valueOf("Method");
+  private static final QName FIELD_TAG = QName.valueOf("Field");
+
+  /** Writes {@code linkageErrors} as exclusion rules into {@code outputFile}. */
+  static void write(Path outputFile, Multimap<SymbolProblem, ClassFile> linkageErrors)
+      throws IOException, XMLStreamException, TransformerException {
+
+    ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+    XMLEventWriter writer = null;
+    try {
+      writer = XMLOutputFactory.newInstance().createXMLEventWriter(buffer);
+
+      writer.add(eventFactory.createStartDocument());
+      writer.add(eventFactory.createStartElement(LINKAGE_CHECKER_FILTER_TAG, null, null));
+
+      for (SymbolProblem symbolProblem : linkageErrors.keySet()) {
+        for (ClassFile classFile : linkageErrors.get(symbolProblem)) {
+          writeXmlEvents(writer, symbolProblem, classFile);
+        }
+      }
+
+      writer.add(eventFactory.createEndElement(LINKAGE_CHECKER_FILTER_TAG, null));
+      writer.add(eventFactory.createEndDocument());
+
+    } finally {
+      if (writer != null) {
+        writer.close();
+      }
+    }
+
+    try (OutputStream outputStream = Files.newOutputStream(outputFile)) {
+      insertIndent(new ByteArrayInputStream(buffer.toByteArray()), outputStream);
+    }
+  }
+
+  private static void insertIndent(InputStream inputStream, OutputStream outputStream)
+      throws TransformerException {
+    if (System.getProperty("javax.xml.transform.TransformerFactory") == null) {
```

Comment:
You shouldn't need to do this. It should be enough to set that system property to point to the class you want.

Reply:
Removed the unnecessary logic.